src/bsp/src/posix/src/InterlocManager.cpp

```diff
@@ -0,0 +1,148 @@
+#include "InterlocManager.h"
+#include "BSP.h"
+#include <Task.h>
+#include <bsp/BSPContainer.h>
+#include <gazebo_msgs/ModelStates.h>
+#include <regex>
+#include <tf2/LinearMath/Transform.h>
+#include <tf2_geometry_msgs/tf2_geometry_msgs.h>
+
+#define ROBOT_CENTER_SUFFIX "/base_link"
+#define HIVEBOARD_SUFFIX "/hiveboard"
+
+InterlocManager::InterlocManager(ILogger& logger) :
+    m_logger(logger),
+    m_tfListener(m_tfBuffer),
+    m_positionUpdateCallback(nullptr),
+    m_positionUpdateContext(nullptr) {}
+
+tf2::Transform InterlocManager::computeRelativeTransform(
+    const geometry_msgs::Pose& currentAgentPose,
+    const geometry_msgs::TransformStamped& currentAgentHBTransform,
+    const geometry_msgs::Pose& distantAgentPose,
+    const geometry_msgs::TransformStamped& distantAgentHBTransform) {
+
+    // Convert the poses into the HB frame
+    geometry_msgs::Pose currentAgentPoseHBFrame;
+    geometry_msgs::Pose distantAgentPoseHBFrame;
+
+    tf2::doTransform(currentAgentPose, currentAgentPoseHBFrame, currentAgentHBTransform);
+    tf2::doTransform(distantAgentPose, distantAgentPoseHBFrame, distantAgentHBTransform);
```

Comment:
If its not so heavy, you could add in which frame each pose is in. This helps the reader know if tranformation can be done. For example the AgentHBTransform let transform a Pose from the robot Frame into the Hiveboard frame. So adding the suffix RobotFrame after the currentAgentPose so developper have an easier time understanding the process of the function. This could be add a comment in the .h though

Reply:
Actually from the ModelState Message we have the pose the in world frame. Maybe I'm missing something but should be the currentAgentPose inverted?